stdlib/Std/src/Foreign/C/Value.luna

```diff
@@ -31,6 +31,41 @@ class CString:
 
     def free: self.ptr.free
 
+class Array a:
+    Array a
+    ArrayVal Int (Pointer a)
+
+    def ptr: case self of
+        Array a: Pointer a
+        ArrayVal _ p: p
+
+    def size: case self of
+        ArrayVal s _: s
+
+    def make: ArrayVal
+
+    def alloc size:
+        ptr = self.ptr.mallocElems size
+        ArrayVal size ptr
+
+    def free:
+        self.ptr.free
+
+    def readAt index: self . ptr . moveElems index . read
```

Comment:
Should we provide functions that check array bounds?

Reply:
My instinct would be to default to bounds checked, as we want to provide a nicer interface to the underlying C.

Reply:
I think default bounds checking is the way to go